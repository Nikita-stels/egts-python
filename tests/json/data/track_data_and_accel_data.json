{
  "transport": {
    "pt": 1
  },
  "service": {
    "sdr": [
      {
        "rfl": {
          "ssod": 0,
          "rsod": 0
        },
        "tm": "0x0fd90000",
        "sst": 10,
        "rst": 10,
        "rd": [
          {
            "srt": 62,
            "srd": {
              "atm": 300000,
              "tds": [
                {
                  "flags": {
                    "rtm": 10
                  },
                  "lat": 0,
                  "long": 0,
                  "spd": 0,
                  "dir": 100
                },
                {
                  "flags": {
                    "rtm": 11
                  },
                  "lat": 56.0003,
                  "long": 44.0025,
                  "spd": 1000,
                  "dir": 120
                }
              ]
            }
          },
          {
            "srt": 20,
            "srd": {
              "atm": "0x02ADC4F6",
              "ads": [
                {
                  "rtm": 0,
                  "xaav": 0,
                  "yaav": 0,
                  "zaav": 0
                },
                {
                  "rtm": 1,
                  "xaav": 1,
                  "yaav": 1,
                  "zaav": 1
                }
              ]
            }
          }
        ]
      }
    ]
  }
}
