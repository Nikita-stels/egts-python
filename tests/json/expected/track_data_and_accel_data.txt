EGTS protocol validator v.1.0  Copyright JSC "Navigation-information systems", 2011
------------------------------------------------------------------------------------
File: 'D:\PycharmProjects\egts_protocol\tests\json\generated\tmp_msg.bin'
Packet data: 
 0100000B004300010001E638000100040000D90F0A0A3E1D0002E0930400EA0000000000000000000064EB8B2C4A9F42D2943EE8037814150002F6C4AD02000000000000000001000100010001000411


EGTS Transport Layer:
---------------------
  Validating result   - 0 (OK)

  Protocol Version    - 1
  Security Key ID     - 0
  Flags               - 00000000b (0x00)
       Prefix         - 00
       Route          -   0
       Encryption Alg -    00
       Compression    -      0
       Priority       -       00 (the highest)
  Header Length       - 11
  Header Encoding     - 0
  Frame Data Length   - 67
  Packet ID           - 1
  No route info       -
  Header Check Sum    - 0xE6

EGTS Service Layer:
---------------------
  Validating result   - 0 (OK)

  Packet Type         - EGTS_PT_APPDATA
  Service Layer CS    - 0x1104

    Service Layer Record:
    ---------------------
    Validating Result    - 0 (OK)

    Record Length               - 56
    Record Number               - 1
    Record flags                -     00000100b (0x04)
        Sourse Service On Device    - 0
        Recipient Service On Device -  0
        Group Flag                  -   0
        Record Processing Priority  -    00 (the highest)
        Time Field Exists           -      1
        Event ID Field Exists       -       0
        Object ID Field Exists      -        0
    Time                        - 0x0FD90000 (05.06.2018 07:25:52 GMT)
    Source Service Type         - 10 (EGTS_ECALL_SERVICE) 
    Recipient Service Type      - 10 (EGTS_ECALL_SERVICE) 

       Subrecord Data:
       ------------------
       Validating Result   - 0 (OK)

       Subrecord Type      - 62 (EGTS_SR_ECALL_TRACK_DATA)
       Subrecord Length    - 29
       Structures Amount      - 2
       Absolute Time          - 0x000493E0 (04.01.2010 11:20:00.000  GMT)
       Track Points Data Structures:
       -----------------------------
                              1
           Relative Time      - [1000 ms] (04.01.2010 11:20:01.000 GMT)
             Point Data Exist - 1
             Long. Hemisphere -  1
             Lat.  Hemisphere -   1
           LAT                - 0x00000000 (N0.00000000)
           LON                - 0x00000000 (W0.00000000)
           SPEED              - 0  (0.00 kmph)
           DIR                - 100 
           ------------------
                              2
           Relative Time      - [1100 ms] (04.01.2010 11:20:02.100 GMT)
             Point Data Exist - 1
             Long. Hemisphere -  1
             Lat.  Hemisphere -   1
           LAT                - 0x9F4A2C8B (N56.00030000)
           LON                - 0x3E94D242 (W44.00249997)
           SPEED              - 1000  (10.00 kmph)
           DIR                - 120 
           ------------------

       Subrecord Data:
       ------------------
       Validating Result   - 0 (OK)

       Subrecord Type      - 20 (EGTS_SR_ACCEL_DATA)
       Subrecord Length    - 21
       Structures Amount      - 2
       Absolute Time          - 0x02ADC4F6 (05.06.2011 04:03:02.000 GMT)
       Accelerometr Data Structures:
       -----------------------------
                              1
           Relative Time      - [000 ms] (05.06.2011 04:03:02.000 GMT)
           XAAV               - 0x0000  0.0m/c2
           YAAV               - 0x0000  0.0m/c2
           ZAAV               - 0x0000  0.0m/c2

           ------------------
                              2
           Relative Time      - [001 ms] (05.06.2011 04:03:02.001 GMT)
           XAAV               - 0x0001  0.1m/c2
           YAAV               - 0x0001  0.1m/c2
           ZAAV               - 0x0001  0.1m/c2

           ------------------


-------------------------------------------------------------
Press any key to exit...
